File: python/Singleton/empty/SingletonEager.py
class SingletonEager:
    obj = None

    #initialization of static variable to owner class not allowed in python
    #so we do it in constructor
    def __init__(self):
        #instantiate class on init and set log to blank

    @staticmethod
    def get_instance():
        #return the single instance

    def add_to_log(self, string):
        #append "string" to the current log

    def get_log(self):
        #return the current log

File: java/Singleton/empty/SingletonEager.java

public class SingletonEager {

  //create singleton object

  private String log;

  private SingletonEager(){
    //set the log to blank
  }

  public static SingletonEager get_instance(){
    //return the Singleton object
  }

  public void add_to_log(String s){
    //add String s to the current log - pay attention to newlines
  }

  public String get_log(){
    //return the current log
  }
}
